libs/pdi-scanner/src/main.rs

```diff
@@ -0,0 +1,349 @@
+use clap::Parser;
+use image::EncodableLayout;
+use std::{
+    io::{self, Write},
+    process::exit,
+    time::Duration,
+};
+use tracing_subscriber::prelude::*;
+
+use base64::{engine::general_purpose::STANDARD, Engine as _};
+use pdi_scanner::{
+    client::Client,
+    connect,
+    protocol::{
+        image::{RawImageData, Sheet},
+        packets::Incoming,
+        types::{
+            DoubleFeedDetectionCalibrationType, DoubleFeedDetectionMode, EjectMotion, ScanSideMode,
+            Status,
+        },
+    },
+    scanner::{Scanner, StopMode},
+};
+
+#[derive(Debug, Parser)]
+struct Config {
+    #[clap(long, env = "LOG_LEVEL", default_value = "warn")]
+    log_level: tracing::Level,
+}
+
+fn setup(config: &Config) -> color_eyre::Result<()> {
+    color_eyre::install()?;
+    setup_logging(config)?;
+    Ok(())
+}
+
+fn setup_logging(config: &Config) -> color_eyre::Result<()> {
```

Comment:
The logging uses a very flexible crate called `tracing`, and we set it up in this function to go to `stderr` because we're otherwise using `stdout` as one half of a communication channel.

Reply:
Could you add some instructions for turning on the debug logging to the README?